doc/user/content/concepts/indexes.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Indexes
+description: Conceptual page on indexes.
+menu:
+  main:
+    parent: concepts
+    weight: 20
+    identifier: 'concepts-indexes'
+---
+
+## Overview
+
+Indexes can improve query performance.
+
```

Comment:
Updated this page to focus on indexes and views (instead of arrangements and dataflows).

Can add a bit more about general query performance - and link to the optimization page.  I want to make sure that people are aware of the improvements indexes provide, in addition to the in-memory.